### PR TITLE
DCOS-14630: Split up artifacts reducer

### PR DESCRIPTION
**Pull request by Poltergeist**
**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

Move FormReducer, JSONReducer/Parser to
their own file, being `FormReducer/Artifacts.js` and `JSONReducer/Artifacts.js`.

Close DCOS-14630

### CODE REVIEW

File: plugins/services/src/js/reducers/JSONMultiContainerReducers.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ module.exports = {
       }
     };
   },
-  fetch,
```
Comment (nLight) [inline plugins/services/src/js/reducers/JSONMultiContainerReducers.js]: so was it a bug?

File: plugins/services/src/js/reducers/serviceForm/FormReducers/Artifacts.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import {
+  ADD_ITEM,
+  REMOVE_ITEM,
+  SET
+} from '../../../../../../../src/js/constants/TransactionTypes';
+
+function processTransaction(state, {type, path, value}) {
```
Comment (nLight) [inline plugins/services/src/js/reducers/serviceForm/FormReducers/Artifacts.js]: Do we duplicate the code 🚨 ?

Comment (Poltergeist) [inline plugins/services/src/js/reducers/serviceForm/FormReducers/Artifacts.js]: You mean duplication of the code in the JSON reducer? Well I had an interessting discussion about that the other day. Basically I was also thinking this is duplication, but it actually isn't. With one function we are creating the json output and with the other one we are creating the Form output. In this case these both happen to be the same. So if one changes the other one will still work.

Comment (nLight) [inline plugins/services/src/js/reducers/serviceForm/FormReducers/Artifacts.js]: ok, that makes sense

**Comment by jfurrow** [comment]
Nice work @Poltergeist!

**Pull request merged**